### ISSUE TITLE
BUG: QuarterBegin doesn't match other offsets for n=0

### ISSUE THREAD

**Issue by chris-b1** [pydata/pandas#11406]
xref #8435 (different issue though)
normally the semantic is to roll forward to the next period for `n=0`

```python
In [15]: pd.Timestamp('2014-3-2') + pd.offsets.MonthBegin(n=0)
Out[15]: Timestamp('2014-04-01 00:00:00')

In [16]: pd.Timestamp('2014-3-2') + pd.offsets.QuarterBegin(n=0)
Out[16]: Timestamp('2014-03-01 00:00:00')

In [17]: #expected: Timestamp('2014-06-01 00:00:00')
```

**Comment by jreback** [comment]
I recall this from a discussion within one of the freq issues (though don't remember the substance)
maybe have a look thru some of the issues

**Comment by chris-b1** [comment]
haven't found much yet, but I guess this behavior was explicitly tested - 
https://github.com/pydata/pandas/blob/master/pandas/tseries/tests/test_offsets.py#L2921

**Comment by jreback** [comment]
@chris-b1 yeh at the very least quarters should be consistent. I'd say go with how month/year work. Make a nice big note in what's new / docs w.r.t. this as its an API breaking change.

**Comment by jreback** [comment]
closed by #11777
